stripe/resource_stripe_promotion_code.go

```diff
@@ -132,21 +132,32 @@ func resourceStripePromotionCodeCreate(ctx context.Context, d *schema.ResourceDa
 		if err != nil {
 			return diag.FromErr(err)
 		}
-		params.ExpiresAt = stripe.Int64(t.Unix())
+		expiresAtUnixStripeValue := stripe.Int64(t.Unix())
```

Comment:
I have a feeling this block can be rewritten by leveraging https://pkg.go.dev/time#Time.IsZero into:

`
if t.IsZero() {
  params.ExpiresAt = stripe.Int64(t.Unix())
}
`

Reply:
The formatting is scattered by GitHub (the if condition should be in the block not inline).

Reply:
Done!